app/controllers/participants/validations_controller.rb

```diff
@@ -0,0 +1,224 @@
+  class ValidationsController < BaseController
+    skip_before_action :ensure_participant, only: :reset
+    before_action :set_form
+    before_action :check_not_already_completed, except: %i[complete reset]
+    before_action :validate_request_or_render, only: %i[do_you_know_your_trn
+                                                        have_you_changed_your_name
+                                                        confirm_updated_record
+                                                        name_not_updated
+                                                        tell_us_your_details
+                                                        confirm_details
+                                                        cannot_find_details]
+
+    def start
+      store_form_and_redirect_to_step :do_you_know_your_trn
+    end
+
+    def do_you_know_your_trn
+      choice = @participant_validation_form.do_you_know_your_trn_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :have_you_changed_your_name
+      elsif choice == "no"
+        store_form_and_redirect_to_step :find_your_trn
+      else
+        store_form_and_redirect_to_step :get_a_trn
+      end
+    end
+
+    def find_your_trn; end
+
+    def get_a_trn; end
+
+    def have_you_changed_your_name
+      choice = @participant_validation_form.have_you_changed_your_name_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :confirm_updated_record
+      else
+        store_form_and_redirect_to_step :tell_us_your_details
+      end
+    end
+
+    def confirm_updated_record
+      choice = @participant_validation_form.updated_record_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :tell_us_your_details
+      elsif choice == "no"
+        store_form_and_redirect_to_step :name_not_updated
+      else
+        store_form_and_redirect_to_step :check_with_tra
+      end
+    end
+
+    def name_not_updated
+      choice = @participant_validation_form.name_not_updated_choice
+      if choice == "register_previous_name"
+        store_form_and_redirect_to_step :tell_us_your_details
+      else
+        store_form_and_redirect_to_step :change_your_details_with_tra
+      end
+    end
+
+    def change_your_details_with_tra; end
+
+    def check_with_tra; end
+
+    def tell_us_your_details
+      store_form_and_redirect_to_step :confirm_details
+    end
+
+    def confirm_details
+      validate_participant_details_and_redirect
+    end
+
+    def cannot_find_details
+      store_validation_data!
+      reset_form_data
+      store_form_and_redirect_to_step :complete
+    end
+
+    def complete
+      @school = participant_profile.school
+      if participant_profile.ecf_participant_eligibility&.eligible_status?
+        # TRN has been validated, qts, no flags, not done before, no previous induction
+        render_completed_page
+      else
+        # all other cases
+        render_manual_check_page
+      end
```

Comment:
I thought there were three possible outcomes here?

Reply:
I was under the impression that all the "holding pen" logic was on the eligibility record - is that not the case?

Reply:
Currently all outcomes result in seeing the manual check page. When there isn't a holding pen, 1 outcome will result in complete page and 2 will result in manual check page.  Holding pen logic is on the eligibility record, just everything ends up on the same page.